[FIX #112] Guile Evaluation tests upon version, using bytecode if > 2…

turbopape opened the pull request:
….0.X (2.1.X etc...)
This makes evaluation works for Guile 2.1.X, and so addresses #112

jaor commented on scheme/guile/geiser/evaluation.scm:
```diff
@@ -70,14 +71,24 @@
+                     (let* ((min-ver (string->number
+				      (cadr
+				       (string-split
+					(version) #\.))))
```
this is just `(minor-version)`

jaor commented on scheme/guile/geiser/evaluation.scm:
```diff
@@ -70,14 +71,24 @@
+						  ((> min-ver 0) 'bytecode)
+						  (else  'objcode)))
+			    (compilation-fn (cond
+					     ((> min-ver 0)  load-thunk-from-memory)
```
instead of repeating the check, we could just have something like `(bytecode? (string=? "0" (minor-version)))` and use that boolean flag.

jaor commented on scheme/guile/geiser/evaluation.scm:
```diff
@@ -70,14 +71,24 @@
+                            (thunk (compilation-fn o)))
+		       (start-stack 'geiser-evaluation-stack
+				    (eval `(,thunk) module))))
+		 (lambda vs vs)))))
```
github is showing changes in whitespace... please make sure you didn't introduce any tabs (maybe you didn't)

turbopape commented:
I addressed you remarks.
I fixed the tabs, used minor version.
But the correct flag is bytecode? is not= "0", as at 2.0.X we still use objcode.
I also rebased.

jaor commented:
hmmm... this change breaks guile 2.0, because `(system vm loader)` is not available (also there's still tabs, but that's minor).  some kind of compile-time cond is needed, perhaps cond-expand, if a feature for 2.1 is provided (that would also substitute the check for minor-version)... investigating...

jaor commented:
@turbopape i've pushed an alternative patch using cond-expand, so i'm closing this PR.  thanks nevertheless for your help!!

turbopape commented:
No problem,as I  am new to guile, I  found this little exchange very
interesting, I learnt a lot, thank you !
Le 14 déc. 2015 02:52, "jao" <notifications@github.com> a écrit :

> @turbopape <https://github.com/turbopape> i've pushed an alternative
> patch using cond-expand, so i'm closing this PR. thanks nevertheless for
> your help!!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jaor/geiser/pull/120#issuecomment-164327976>.
>
